pilot/cmd/istioctl/controlplane/cmd.go

```diff
@@ -0,0 +1,147 @@
+		Long: "istioctl control-plane produces deployment files to run the minimum Istio control for the set of " +
+			"features requested by the --feature flag. If no features are provided, we create deployments for the " +
+			"default control plane: Pilot, Mixer, CA, an initializer for automatic side-car injection, and Ingress " +
+			"Proxies, with mTLS enabled.",
+		Example: `istioctl control-plane --features routing,policy,initializer -o helm`,
+		RunE: func(c *cobra.Command, args []string) error {
+			m.setFeatures(*features)
+			m.Namespace = *istioNamespaceFlag
+			switch strings.ToLower(out) {
+			case "helm":
+				_, err := fmt.Fprint(os.Stdout, valuesFromModel(m))
+				return err
+			case "yaml":
+				rendered, err := yamlFromModel(m, helmChartLocation)
+				if err != nil {
+					return err
+				}
+				_, err = fmt.Fprint(os.Stdout, rendered)
+				return err
+			default:
+				return fmt.Errorf("unsupported output %q", out)
+			}
+		},
+	}
+
+	features = cmd.PersistentFlags().StringArrayP("features", "f", []string{},
+		`List of Istio features to enable. Accepts any combination of "mtls", "telemetry", "routing", "ingress", "policy", "initializer".`)
+	cmd.PersistentFlags().StringVar(&m.Hub, "hub", "gcr.io/istio-testing", "The container registry to pull Istio images from")
+	cmd.PersistentFlags().StringVar(&m.MixerTag, "mixer-tag", "latest", "The tag to use to pull the `mixer` container")
+	cmd.PersistentFlags().StringVar(&m.PilotTag, "pilot-tag", "latest", "The tag to use to pull the `pilot-discovery` container")
+	cmd.PersistentFlags().StringVar(&m.CaTag, "ca-tag", "latest", "The tag to use to pull the `ca` container")
+	cmd.PersistentFlags().StringVar(&m.ProxyTag, "proxy-tag", "latest", "The tag to use to pull the `proxy` container")
+	cmd.PersistentFlags().BoolVar(&m.Debug, "debug", false, "If true, uses debug images instead of release images")
+	cmd.PersistentFlags().Uint16Var(&m.NodePort, "ingress-node-port", 0,
+		"If provided, Istio ingress proxies will run as a NodePort service mapped to the port provided by this flag. "+
+			"Note that this flag is ignored unless the \"ingress\" feature flag is provided too.")
+	cmd.PersistentFlags().StringVarP(&out, "out", "o", "helm", `Output format. Acceptable values are:
+					"helm": produces contents of values.yaml
+					"yaml": produces Kubernetes deployments`)
+
+	// TODO: figure out how we want to package up the charts with the binary to make this easy
+	cmd.PersistentFlags().StringVar(&helmChartLocation, "helm-chart-dir", ".",
+		"The directory to find the helm charts used to render Istio deployments. -o yaml uses these to render the helm chart locally.")
+
+	_ = cmd.PersistentFlags().MarkHidden("hub")
+	_ = cmd.PersistentFlags().MarkHidden("mixer-tag")
+	_ = cmd.PersistentFlags().MarkHidden("pilot-tag")
+	_ = cmd.PersistentFlags().MarkHidden("ca-tag")
+	_ = cmd.PersistentFlags().MarkHidden("proxy-tag")
+	return cmd
+}
+
+type model struct {
```

Comment:
'model' is a little abused word. Maybe call it InstallationConfig? Also please make sure that each field has some default value and users does not need to tweak anything for the default installation case. 

The values.yaml file contains more fields that this struct. Please comment out the unused fields and clean up the yaml files if necessary.

Reply:
Updated, though TBH I don't really like using `installation` here: I think model fits pretty well.

Reply:
Right now the only two options in `values.yaml` that we don't have flags for is using the default config, and using a demo version of the CA.

I plan to rewrite the helm charts as a follow on to this PR, so can we punt till later?